src/components/TransactionStatusToast/TransactionStatusToast.tsx

```diff
@@ -0,0 +1,67 @@
+import { addNewCustomToast, deleteCustomToast } from 'utils';
+import { CustomToast } from 'UI/TransactionsToastList/components/CustomToast';
+import React, { useEffect, useMemo } from 'react';
+import { CustomToastType } from 'types/toasts';
+import { useSelector } from 'reduxStore/DappProviderContext';
+import { customToastsSelector } from 'reduxStore/selectors';
+import { TRANSACTION_STATUS_TOAST_ID } from 'constants/index';
+import styles from './transaction-status-toast.scss';
+import { StatusMessageComponent } from './StatusMessageComponent';
+
+type ErrorToastProps = {
+  type?: 'info' | 'error' | 'warning';
+  message: string;
+  duration?: number;
+  onDelete?: () => void;
+};
+
+export const TransactionStatusToast: React.FC<ErrorToastProps> = ({
+  message,
+  duration = 30000,
+  onDelete,
+  type = 'warning'
+}) => {
+  const customToasts = useSelector(customToastsSelector);
+
+  const errorToast = useMemo(
+    () =>
+      customToasts.find(
+        (customToast: CustomToastType) =>
+          customToast.toastId === TRANSACTION_STATUS_TOAST_ID
+      ),
+    [customToasts]
+  );
+
+  const handleDelete = () => {
+    if (Boolean(errorToast)) {
+      deleteCustomToast(TRANSACTION_STATUS_TOAST_ID);
+      onDelete?.();
```

Comment:
looks like you're running deleteCustom anyway, which means that only `onDelete` could be wrapped inside an if condition

Reply:
good catch